src/utils/featheryClient/index.ts

```diff
@@ -474,6 +515,59 @@ export default class FeatheryClient extends IntegrationClient {
     );
   }
 
+  /**
+   * If there is a pending invocation of submitCustom, this method calls it immediately
+   */
+  flushPendingSubmitCustomUpdates(override: boolean = true) {
+    // we call the debounced method and then flush() to immediately submit changes
+    // see: https://github.com/lodash/lodash/issues/4185#issuecomment-462388355
+    this.debouncedSubmitCustom(override);
+    return this.debouncedSubmitCustom.flush();
+  }
+
+  /**
+   * `beforeunload` event handler that flushes the pending submit custom changes
+   * when a user is attempting to exit the page.
+   * @param event `BeforeUnloadEvent`
+   * @returns
+   */
+  _flushPendingChangesBeforeUnload(event: BeforeUnloadEvent) {
+    event.preventDefault();
+    this.flushPendingSubmitCustomUpdates();
+    return (event.returnValue = '');
+  }
+
+  async submitCustom(
+    customKeyValues: { [key: string]: any },
+    // Options
+    {
+      override = true,
+      shouldFlush = false
+    }: { override?: boolean; shouldFlush?: boolean } = {}
+  ) {
+    if (this.draft || this.noSave) return;
+    if (Object.keys(customKeyValues).length === 0 && !shouldFlush) return;
+    // If there are values passed, aggregate them in the pending queue
+    Object.entries(customKeyValues).forEach(
+      ([key, value]) => (this.pendingCustomFieldUpdates[key] = value)
+    );
+    if (Object.keys(this.pendingCustomFieldUpdates).length) {
```

Comment:
this should happen after hte if statement on line 564 IMO. if there's no debounce no point in adding the unload handler